populus/utils/filesystem.py

```diff
@@ -85,28 +85,24 @@ def is_exe(fpath):
 
 @to_tuple
 def recursive_find_files(base_dir, pattern):
-    for dirpath, _, filenames in os.walk(base_dir):
-        for filename in filenames:
-            if fnmatch.fnmatch(filename, pattern):
-                yield os.path.join(dirpath, filename)
-
+    """
+    Recursively traverse directory tree and find files matching pattern.
 
-@to_tuple
-def find_solidity_source_files(base_dir):
-    return (
-        os.path.relpath(source_file_path)
-        for source_file_path
-        in recursive_find_files(base_dir, "*.sol")
-    )
+    :param base_dir: Base directory to start crawling.
+    :param pattern: File pattern, to match again.
+                    Either string (single pattern match) or tuple (multiple patterns)
+    """
 
+    def match(filename, pattern):
+        if isinstance(pattern, tuple):
+            return any([fnmatch.fnmatch(filename, p) for p in pattern])
+        else:
```

Comment:
I'm thinking this may be better as an `elif is_string(pattern)` and then having an else clause like the following.

```
else:
    raise TypeError("Pattern must either be a string pattern or a list of patterns.  Got {0}".format(pattern))
```

Reply:
Yes I agree, much better!